dev/examples.el

```diff
@@ -28,6 +28,10 @@
 (defun square (num) (* num num))
 (defun three-letters () '("A" "B" "C"))
 
+(defun dash-expand:&hash? (key source)
```

Comment:
@Fuco1 	This looks useful enough to be part of core, move it to `dash.el`

Reply:
Sure.

Reply:
In addition, I think that desctructoring that maps to `map-elt` might be usefull too(or dash reimplementation with similar behaviour). We are going to create such in lsp-mode but I believe it might be usefull for other dash clients.

Reply:
I'm not familiar with `map-elt`, is it some generic kv storage interface?  Does it make sense to use `&map` for it? (in other words, is it generic enough?)

Reply:
Seems like pcase is using `map` matcher and there's `mapp` in core so let's use `&map`.

Reply:
I am not very familiar too, the general idea is that it will `(map-elt m "key")` works with map/plist/alist. I think it is geneneric enough since you could extend it via defmethods to work with other structure(not 100% sure though). I will push another diff with &map implementation.

Reply:
Actually, since `map` is new library it is not present in emacs < 25 do you want this in the core?

Reply:
> I am not very familiar too, the general idea is that it will `(map-elt m "key")` works with map/plist/alist. I think it is geneneric enough since you could extend it via defmethods to work with other structure(not 100% sure though).

`map.el` was added to Emacs 25. It originally supported only alists, hash-tables, and arrays, and its API was not generic.

Emacs 27 reimplements the API as generic functions and adds support for plists. It also deprecates `map-put` and the `testfn` argument of `map-elt` and `map-contains-key`.

Depending on which version of Emacs you're targeting, it may be too early to use `map.el`. See also #179.

Reply:
@yyoncho @basil-conto ok then we can maybe wait.  Or we can create another package `dash-map...` or something :D And put it there.  But that's outside of the scope of this patch.

I think all we need is to add documentation and it's good to go.

Reply:
> Emacs 27 reimplements the API as generic functions and adds support for plists. It also deprecates `map-put` and the `testfn` argument of `map-elt` and `map-contains-key`.

These features were recently released as `map.el` version 2.0, and can be installed even on Emacs 25 and 26 via GNU ELPA: http://elpa.gnu.org/packages/map.html